nuxeo-importer-stream/src/main/java/org/nuxeo/importer/stream/consumer/RedisDocumentMessageConsumer.java

```diff
@@ -0,0 +1,163 @@
+            this.redisPrefix = DEFAULT_REDIS_PREFIX;
+        } else {
+            this.redisPrefix = redisPrefix;
+        }
+        RedisAdmin redisAdmin = Framework.getService(RedisAdmin.class);
+        try {
+            addDocumentSHA = redisAdmin.load("org.nuxeo.importer.stream", "add-document").getBytes("UTF-8");
+        } catch (IOException e) {
+            throw new NuxeoException("Cannot load Redis script", e);
+        }
+        redisExecutor = Framework.getService(RedisExecutor.class);
+    }
+
+    @Override
+    public void accept(DocumentMessage message) {
+        BlobInfo blobInfo = message.getBlobInfo();
+        String blobPath = "";
+        String blobFilename = "";
+        String blobMimeType = "";
+        String properties;
+        try {
+            if (blobInfo == null || StringUtils.isBlank(blobInfo.filename)) {
+                properties = getProperties(message);
+            } else {
+                blobPath = blobInfo.filename;
+                blobMimeType = blobInfo.mimeType;
+                blobFilename = Paths.get(blobInfo.filename).getFileName().toString();
+                properties = getPropertiesWithBlob(message);
+            }
+            String parentPath = message.getParentPath();
+            if (parentPath.startsWith("/")) {
+                parentPath = parentPath.substring(1);
+            }
+            String key = Paths.get(parentPath, message.getName()).toString();
+            String docKey = redisPrefix + ":" + DOC_KEY_SUFFIX;
+            String folderKey = redisPrefix + ":" + FOLDER_KEY_SUFFIX;
+            String dataKey = redisPrefix + ":" + DATA_KEY_SUFFIX + ":" + key;
+            String level = String.valueOf(key.split("/").length);
+            String url = URLEncoder.encode(key, "UTF-8").replaceAll("%2F", "/");
+            String payload = getPayload(message, properties);
+            redisExecutor.evalsha(addDocumentSHA, Arrays.asList(bytes(docKey), bytes(dataKey), bytes(folderKey)),
+                    Arrays.asList(bytes(key), bytes(parentPath), bytes(message.getType()), bytes(message.getName()),
+                            bytes(payload), bytes(url), bytes(level), bytes(blobPath), bytes(blobFilename),
+                            bytes(blobMimeType)));
+        } catch (UnsupportedEncodingException e) {
+            throw new NuxeoException("Cannot urlencode: " + message, e);
+        } catch (JsonProcessingException e) {
+            throw new NuxeoException("Cannot convert properties to json: " + message, e);
+        }
+    }
+
+    protected String getPayload(DocumentMessage message, String properties) {
+        return String.format("{\"entity-type\": \"document\", \"name\": \"%s\", \"type\": \"%s\", \"properties\": %s}",
+                message.getName(), message.getType(), properties);
+    }
+
+    protected String getPropertiesWithBlob(DocumentMessage message) {
+        ObjectMapper mapper = new ObjectMapper();
```

Comment:
You should use a `static` pre-computed `ObjectMapper` instead of creating one each time.